### PR TITLE
fix: focus issue in filterable multiselect

### LINKED ISSUE #16830: [Bug]: Focus on FilterableMultiSelect always persists once click on the control
### Package

@carbon/react

### Browser

Chrome, Firefox

### Package version

1.60.1

### React version

18.2.0

### Description

Hi Team, 

I have upgraded the carbon library in my application and replace the Multiselect drop down to FilterableMultiSelect. I am observing one issue while using this control.

After the page loads, If click on the control, the focus always persists,  if we move to another control by clicking through the mouse or moving one control to another control using tab button

Please look into this issue and provide a solution on this.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/experimental-unstable-slug-examples--filterable-multiselect

### Steps to reproduce

1. Open this URL: https://react.carbondesignsystem.com/?path=/story/experimental-unstable-slug-examples--filterable-multiselect
2. Click on the control
3. Next click outside of the control
4. The focus of the control remains there. it should be removed from the control
image

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Application/PAL

https://react.carbondesignsystem.com/?path=/story/experimental-unstable-slug-examples--filterable-multiselect

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by riddhybansal**
Closes #16830 

Focus issue in filterable multiselect

#### Changelog

**New**

- Added onBlur

#### Testing / Reviewing

Test filterable multiselect , it should work as intended. 
Using the Storybook Filterable, if you click in the select and type opt ... then try clicking on the (x) clear button, it should work , and it should lose focus after clicking outside.

Please verify this PR does not create another issue that has been closed already #16145

### CODE REVIEW

**Comment by riddhybansal** [comment]
My only concerned over here is , is this the intended behaviour - when the filterable multiselect is opened clicking outside first closes the menu , and focus remains on input field to type something and then clicking outside again clears the focus.

**Comment by Gururajj77** [comment]
@riddhybansal , clicking the (x) button isn't clearing the text box or clearing the selected options, is that intended behaviour?? Please correct me if I missed something

https://github.com/carbon-design-system/carbon/assets/89023023/3bd0a602-561d-4fba-a903-bfbb3db263e8

**Comment by alisonjoseph** [comment]
I'm also seeing the same issue as @Gururajj77

**Comment by guidari** [comment]
> My only concerned over here is , is this the intended behaviour - when the filterable multiselect is opened clicking outside first closes the menu , and focus remains on input field to type something and then clicking outside again clears the focus.

Also looks like the focus doesn't go away when we are tabbing out from the `ListBoxMenu`, maybe if we add an onBlur there it would fix your comment and the tab navigation...

> @riddhybansal , clicking the (x) button isn't clearing the text box or clearing the selected options, is that intended behaviour?? Please correct me if I've missed something

Good catch. In the [1.59.0](https://deploy-preview-16696--v11-carbon-react.netlify.app/?path=/story/components-multiselect--filterable) is working as expected the close icon.

We should also write tests for those two scenarios, to avoid any changes that might affect that in the future.

File: packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx [hunk 1]
```diff
@@ -714,6 +712,11 @@ const FilterableMultiSelect = React.forwardRef(function FilterableMultiSelect<
         }
       },
       onFocus: () => setInputFocused(true),
+      onBlur: () => {
+        !isOpen && setInputFocused(false);
+        setInputValue('');
+        // setIsOpen(false);
```
Comment (tay1orjones) [inline packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx]:
```suggestion
```

**Review by tay1orjones — APPROVED** [review]

**Review by guidari — APPROVED** [review]

**Pull request merged**